File: rb-appscript/trunk/sample/Print_folder_tree.rb
#!/usr/bin/env ruby

# Prints the sub-folder hierarchy of a given folder as a list of folder names indented according to depth.

require "appscript"
include Appscript

def print_folder_tree(folder, indent='')
    puts indent + folder.name.get
    folder.folders.get.each { |folder| print_folder_tree(folder, indent + "\t") }
end

print_folder_tree(app('Finder').home.folders['Documents'])

File: py-appscript/trunk/Examples/appscript/Print_folder_tree.py
#!/usr/bin/env python

# Prints the sub-folder hierarchy of a given folder as a list of folder names
# indented according to depth.

from appscript import *


def printFolderTree(folder, indent=''):
    """Print a tab-indented list of a folder tree."""
    print indent + folder.name.get().encode('utf8')
    for folder in folder.folders.get():
        printFolderTree(folder, indent + '\t')


printFolderTree(app('Finder').home.folders['Documents'])